extensions/smallrye-metrics/deployment/src/main/java/io/quarkus/smallrye/metrics/deployment/SmallRyeMetricsProcessor.java

```diff
@@ -165,13 +167,13 @@ void createRoute(BuildProducer<RouteBuildItem> routes,
         }
         routes.produce(new RouteBuildItem.Builder()
                 .routeFunction(route)
```

Comment:
Would it be better for `NonApplicationRootPathBuildItem.adjustPath()` to do this adjustment rather than sprinkling it into each extension that needs it? As I think all extensions would need this.

I vaguely recall having it do that at one point but can't remember why I took it out now. Maybe because at the time I was thinking /q would always be at the root and not be impacted by the HTTP root.

One thing we need to think about is how this being done at static init is impacted by the root path being changed on the command line at start. That might have been the reasons why I was originally thinking not hanging it off the app path

Reply:
> Would it be better for NonApplicationRootPathBuildItem.adjustPath() to do this adjustment rather than sprinkling it into each extension that needs it? 

Perhaps yes, but then we would need to refactor these two build items and probably merge them into one, because I think you can't inject a build item into another build item to read its value... We would need a single build item that knows both of these configured roots.

To make it more complicated, in some places you need to pass a path adjusted only for the `nonApplicationRootPath` (like the constructor of `NotFoundPageDisplayableEndpointBuildItem` for example), while other places require a path adjusted for both root paths (like here in this PR), and some places require the path to be not adjusted at all (like `RouteBuildItem.route(path)`).
Maybe it would make more sense to improve the javadocs to make it more clear how adjusted a path needs to be?

> One thing we need to think about is how this being done at static init is impacted by the root path being changed on the command line at start.

It's not impacted at all, overriding the root paths at start has no effect, because they're build-time-only properties (you get a warning if you try to override them at startup).

Reply:
Shouldn't be an issue to modify https://github.com/quarkusio/quarkus/blob/master/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java#L83 to take the `HttpBuildPathItem` as a parameter and pass it to the creation of the non application one.

Could we have two methods for adjusting the path on `NonApplicationRootPathBuildItem`? One that takes into account the HTTP Root and one that doesn't?

For some reason I thought it was run time, must have been confused with port.

Reply:
It feels a little bit like duplicating functionality of one BuildItem in another, but ok I applied that, please check